editor/test/integration/test_util.clj

```diff
@@ -529,8 +529,12 @@
   (get-in (g/node-value node-id :_properties) [:properties label :error]))
 
 (defn prop! [node-id label val]
-  (let [[node-id label] (resolve-prop node-id label)]
-    (g/set-property! node-id label val)))
+  (let [prop (get-in (g/node-value node-id :_properties) [:properties label])]
+    (if-let [set-fn (-> prop :edit-type :set-fn)]
+      (g/transact
+        (set-fn (g/make-evaluation-context) node-id (:value prop) val))
```

Comment:
You should probably get the `tx-data` from the `set-fn` inside a `(g/with-auto-evaluation-context)` expression so that any reads are cached, then call `(g/transact tx-data)`.

Reply:
Hmm, I only repeated what `properties/set-values!` does (it uses a throw-away evaluation context), not sure why it was done like that, maybe the expectation was that whatever enters the cache will be thrown away by the following transaction?

Reply:
It really depends on what you evaluate inside the `set-fn`. It is not unheard of to evaluate something upstream of the property you're setting.

Reply:
Okay, let's do this!